ReactSkia/components/RSkComponentImage.cpp

```diff
@@ -127,5 +137,67 @@ RnsShell::LayerInvalidateMask RSkComponentImage::updateComponentProps(const Shad
     }
     return updateMask;
 }
+
+sk_sp<SkImage> RSkComponentImage::getImageData(const char *path) {
+  sk_sp<SkImage> imageData{nullptr};
+  sk_sp<SkData> data;
+  if(!path) {
+    RNS_LOG_ERROR("Invalid File");
+    return nullptr;
+  }
+  data = SkData::MakeFromFileName(path);
+   if (!data) {
+    RNS_LOG_ERROR("Unable to make SkData for path : " << path);
+    return nullptr;
+  }
+  imageData = SkImage::MakeFromEncoded(data);
+  return imageData;
+}
+
+void RSkComponentImage::drawAndSubmit() {
+  layer()->client().notifyFlushBegin();
+  layer()->invalidate( RnsShell::LayerPaintInvalidate);
+  if (layer()->type() == RnsShell::LAYER_TYPE_PICTURE) {
+    RNS_PROFILE_API_OFF(getComponentData().componentName << " getPicture :", static_cast<RnsShell::PictureLayer*>(layer().get())->setPicture(getPicture()));
+  }
+  layer()->client().notifyFlushRequired();
+}
+
+void RSkComponentImage::getUriImage(ImageSource source) {
+  struct RemoteImageData* remoteImageData(new struct RemoteImageData);
+  auto sharedCurlNetworking = CurlNetworking::sharedCurlNetworking();
+  CurlRequest *curlRequest = new CurlRequest(nullptr,source.uri.c_str(),0,"GET");
+
+  // callback for remoteImageData
+  remoteImageData->callback = [&](const char* path, char* response, int size) {
+    // Responce callback from network. Get image data, insert in Cache and call Onpaint
+    sk_sp<SkData> data = SkData::MakeWithCopy(response,size);
+    sk_sp<SkImage> imageData = SkImage::MakeFromEncoded(data);
```

Comment:
data check should be added.
If MakeFromEncoded is handling the null check of data , then its okay.